src/microbe_stage/systems/ProcessSystem.cs

```diff
@@ -404,11 +409,11 @@ public static Dictionary<Compound, CompoundBalance> ComputeCompoundBalance(
     ///     input compounds present in <see cref="biome"/>)
     ///   </para>
     /// </remarks>
-    public static Dictionary<Compound, CompoundBalance> ComputeCompoundBalanceAtEquilibrium(
+    public static void ComputeCompoundBalanceAtEquilibrium(
         IEnumerable<OrganelleDefinition> organelles, IBiomeConditions biome, CompoundAmountType amountType,
-        EnergyBalanceInfo energyBalance)
+        EnergyBalanceInfo energyBalance, Dictionary<Compound, CompoundBalance>? result)
     {
-        var result = new Dictionary<Compound, CompoundBalance>();
+        result ??= new Dictionary<Compound, CompoundBalance>();
```

Comment:
Same problem with the ref here, for clarity I would make the result in this method chain non-nullable because it doesn't make that much sense why a caller wouldn't be able to allocate a dictionary for the results.